include/clangmetatool/meta_tool_factory.h

```diff
@@ -53,6 +58,34 @@ class MetaToolFactory : public clang::tooling::FrontendActionFactory {
 #else
   virtual clang::FrontendAction *create() { return new T(replacements, args); }
 #endif
+
+  /**
+   * Run the tool passed in and export any replacements to a YAML file
+   */
+  int runAndExportChanges(clang::tooling::ClangTool &tool,
+                           llvm::raw_ostream &os) {
+    int r = tool.run(this);
+    if (r) {
+      return r;
+    }
+    auto mainFileIt = replacements.begin();
+    // Like clang-tidy's --export-replacements, use the first file passed into
+    // the command line invocation as the 'main' file
+    if (mainFileIt == replacements.end()) {
+      return r;
+    }
+    clang::tooling::TranslationUnitReplacements TUR;
+    TUR.MainSourceFile = mainFileIt->first;
```

Comment:
I think this is intended to be the main source file passed in to the compiler, not the first file with replacements. That being said I'm not sure it overly matters, clang-apply-replacements seems to ignore this value, so maybe its not worth doing anything more complicated?

Reply:
Eg: https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py#L106

So maybe just setting to "" is fine too?

Reply:
My guess is that this field is deprecated

Reply:
Ah yeah, looks like it's not necessary. Removing it